### DISCUSSION TITLE
How to migrate preexisting borg backups when switching to remote borg backup?

### DISCUSSION THREAD

**Discussion by michaelfliegner** [nextcloud/all-in-one#5593] [Questions] [answered]
As of now I can only configure a new instance for backup to BorgBase.
Switching from preexisting  local backups I would have to borg transfer my local backups to BorgBase. Thereafter I could remote restore from the new instance. But restoring the instance from a preexisting archive implies returning to local backups, right? 

How to handle that?
Shouldn't the solution be extended to include
- import of preexisting archives and
- change of backup location?

Then restoring from a preexisting archive would entail only 
- upgrading to AIO >= v10
- toggle back to remote backup.

**Comment by szaimen** [comment, marked as answer]
@timdiels do you have some ideas how to do the above?

**Reply by hannahdiels** [reply]
I'd suggest adding instructions to the readme on manually moving the local repo to the remote and then allow resetting the backup location (already part of the UI but gets hidden after the first backup is made I think).

**Reply by hannahdiels** [reply]
Some complications to take into account such as disabling automatic backups when resetting it.

**Reply by michaelfliegner** [reply]
so we're actually getting somewhere quite soon!

**Reply by archont00** [reply]
I have my own borg backup server. That's how I migrated from local repo path to remote repo path:
1. Stop all containers (from AIO interface).
2. Move `borg.config` (in nextcloud-aio-mastercontainer volume) to a temporary location.
3. Move all files from existing borg repo location to a new location managed by remote borg server. (I did it manually by moving the files and correcting ownership of all files - because my borg backup server runs on the same machine.)
4. Restore `borg.config` from temporary location back to nextcloud-aio-mastercontainer volume).
5. Re-create AIO master container (in my case: `docker-compose down`, change to "beta" channel to get v10.0.0 and `docker-compose up -d`)
6. In AIO interface, click on `Reset backup location` and enter the remote URL.
7. In AIO interface, click on `Create backup` - it fails due to authentication error. Copy the displayed SSH pub key to the borg backup server's `authorized_keys`.
8. In AIO interface, click on `Create backup` again. It should work now.

I did not test to restore the remote backup.

P.S. It would be good if the entry field in AIO interface for remote URL displays `ssh://user@host:port/path_to_repo` hint.

EDIT: Sorry, I have misunderstood the title of the issue `How to migrate preexisting borg backups when switching to remote borg backup?`. Yet it might help some to solve a similar use case as mine.

**Reply by michaelfliegner** [reply]
```
I did not test to restore the remote backup.
```

My question was just that: what happens after restoring from remote?

And it has been answered by @timdiels .

**Reply by hannahdiels** [reply]
Ah one other thing, we need a way to set the encryption password when resetting in this case. I think now it just generates a new password which doesn't match the existing repo's password.

**Reply by hannahdiels** [reply]
Although we already know the password, if it just keeps using the existing one after we reset, we should be fine

**Reply by KoMa1012** [reply]
>     6. In AIO interface, click on `Reset backup location` and enter the remote URL.

I can't find `Reset backup location` in my nextcloud aio interface, I stopped the containers (except master obviously) but it's still not there. Is there anything special I have to do? I'm on 10.0.0 (main channel).

**Reply by asavageiv** [reply]
FYI I was able to get this to work by moving step 4 to after Step 6. `Reset backup location` only appears if `borg.config` is not present. 

The updated steps are:

1. Stop all containers (from AIO interface).
1. Move borg.config (in nextcloud-aio-mastercontainer volume) to a temporary location.
1. Move all files from existing borg repo location to a new location managed by remote borg server. (I did it manually by moving the files and correcting ownership of all files - because my borg backup server runs on the same machine.)
1. Re-create AIO master container (in my case: `docker compose down; docker compose up -d`)
1. In AIO interface, click on Reset backup location and enter the remote URL.
1. Restore borg.config from temporary location back to nextcloud-aio-mastercontainer volume).
1. In AIO interface, click on Create backup - it fails due to authentication error. Copy the displayed SSH pub key to the borg backup server's authorized_keys.
1. In AIO interface, click on Create backup again. It should work now.

I also didn't need to do `Step 3` because my previous location was an sshfs mount. These steps allowed me to create a backup at the new location without moving the repository, which would have been slow because it is quite large.

**Reply by Croydon** [reply]
I successfully moved my local borg backups to a remote server by doing something similar as @asavageiv and @archont00 have done. I did not re-create the AIO master container however, and it seems to work just fine.

**Reply by mannte** [reply]
If you forgot to make a copy of your config file. You can copy `YourBorgBackupLocation/borg/config` to `/var/lib/docker/volumes/nextcloud_aio_mastercontainer/_data/data/borg.config` 
Then set the permissions to
```
chown 0:0 /var/lib/docker/volumes/nextcloud_aio_mastercontainer/_data/data/borg.config
chmod 750 /var/lib/docker/volumes/nextcloud_aio_mastercontainer/_data/data/borg.config
```

**Comment by jakobroehrl** [comment]
For a hetzner storage box with sub-account the remote borg repo must be specified so: 
ssh://u123456-sub1@u123456.your-storagebox.de:23/home/u123456-sub1
